### ISSUE TITLE
error processing PVC "openshift-ansible-service-broker"/"etcd": PVC openshift-ansible-service-broker/etcd has non-bound phase ("Pending") or empty pvc.Spec.VolumeName ("")

### ISSUE THREAD

**Issue by runcom** [openshift/openshift-ansible#6296]
openshift-ansible latest master

Installing openshift (enterprise) with CRI-O fails to create the ansible-service-broker:
```
error processing PVC "openshift-ansible-service-broker"/"etcd": PVC openshift-ansible-service-broker/etcd has non-bound phase ("Pending") or empty pvc.Spec.VolumeName ("")
```

```
[root@rhel7-4-a ~]# oc get pvc -n openshift-ansible-service-broker
NAME      STATUS    VOLUME    CAPACITY   ACCESS MODES   STORAGECLASS   AGE
etcd      Pending                                                      7m
[root@rhel7-4-a ~]# oc describe pvc etcd -n openshift-ansible-service-broker
Name:          etcd
Namespace:     openshift-ansible-service-broker
StorageClass:
Status:        Pending
Volume:
Labels:        <none>
Annotations:   <none>
Capacity:
Access Modes:
Events:
  Type    Reason         Age               From                         Message
  ----    ------         ----              ----                         -------
  Normal  FailedBinding  1m (x26 over 7m)  persistentvolume-controller  no persistent volumes available for this claim and no storage class is set
```
```
[root@rhel7-4-a ~]# oc get pvc -n openshift-ansible-service-broker
NAME      STATUS    VOLUME    CAPACITY   ACCESS MODES   STORAGECLASS   AGE
etcd      Pending                                                      12m
[root@rhel7-4-a ~]# oc get pv -n openshift-ansible-service-broker
No resources found.
```
```
FAILED - RETRYING: Verify that TSB is running (1 retries left).
fatal: [192.168.122.165]: FAILED! => {"attempts": 120, "changed": false, "cmd": ["curl", "-k", "https://apiserver.openshift-template-service-broker.svc/healthz"], "delta": "0:00:01.011412", "end": "2017-11-28 12:17:36.487063", "failed": true, "msg": "non-zero return code", "rc": 7, "start": "2017-11-28 12:17:35.475651", "stderr": "  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current\n                                 Dload  Upload   Total   Spent    Left  Speed\n\r  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0curl: (7) Failed connect to apiserver.openshift-template-service-broker.svc:443; Connection refused", "stderr_lines": ["  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current", "                                 Dload  Upload   Total   Spent    Left  Speed", "", "  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0curl: (7) Failed connect to apiserver.openshift-template-service-broker.svc:443; Connection refused"], "stdout": "", "stdout_lines": []}
```

/cc @mrunalp @giuseppe @sdodson 

we don't have CRI-O errors, I think it's something we're not setting up from the installer, could you help?

**Comment by sdodson** [comment]
@fabianvf For some reason i thought ASB would use emptydir for etcd volume if a storage type wasn't specified, did that break?

**Comment by sdodson** [comment]
@runcom what's the status of the api server in kube-service-catalog namespace?

**Comment by runcom** [comment]
@sdodson I thrown away that cluster but I can reproduce easily, I'll spin up another one and provide status asap

**Comment by runcom** [comment]
@sdodson apiserver is running:
```
[root@rhel7-4-a ~]# oc get pods -n kube-service-catalog
NAME                       READY     STATUS    RESTARTS   AGE
apiserver-kbgr6            1/1       Running   0          11m
controller-manager-nfcc7   1/1       Running   0          11m
[root@rhel7-4-a ~]# oc get pods -n openshift-ansible-service-broker
NAME                READY     STATUS              RESTARTS   AGE
asb-1-48b6q         0/1       CrashLoopBackOff    6          9m
asb-1-deploy        1/1       Running             0          9m
asb-etcd-1-deploy   1/1       Running             0          9m
asb-etcd-1-vzjtq    0/1       ContainerCreating   0          9m
[root@rhel7-4-a ~]# oc describe pvc etcd -n openshift-ansible-service-broker
Name:          etcd
Namespace:     openshift-ansible-service-broker
StorageClass:
Status:        Pending
Volume:
Labels:        <none>
Annotations:   <none>
Capacity:
Access Modes:
Events:
  Type    Reason         Age               From                         Message
  ----    ------         ----              ----                         -------
  Normal  FailedBinding  3m (x26 over 9m)  persistentvolume-controller  no persistent volumes available for this claim and no storage class is set
```

**Comment by runcom** [comment]
> @fabianvf For some reason i thought ASB would use emptydir for etcd volume if a storage type wasn't specified, did that break?

I think the thing is we don't even have any PV at all :confused:

**Comment by fabianvf** [comment]
The current behavior with ASB is that it will not come up without some kind of PV to bind to for etcd. @sdodson we could make it use emptydir by default, I think we avoided doing that earlier because losing the ASB data would cause the service catalog and broker metadata to diverge in a likely unrecoverable way (short of manually going through and deleting all the resources).

**Comment by runcom** [comment]
The same happens with docker, but docker pass the templateservicebroker tests :confused:

**Comment by runcom** [comment]
nevermind, this is a network issue in CRI-O port forwarding

**Comment by cooktheryan** [comment]
@fabianvf and @sdodson sorry to bring this back from the dead but are there any plans to proceed with emptyDir if a default storageclass is not available at  install time?

**Comment by sdodson** [comment]
@cooktheryan I'd have to leave that up to ASB team to decide. Long term they're looking at removing the need to have an etcd instance but I don't think that's going to happen soon.

**Comment by jwmatthews** [comment]
For 3.9 the ASB still uses it's own etcd which requires a PV.

For 3.10+ and going forward the default configuration will store data in CR instances in the broker's namespace, this avoids the dependency on our own etcd and the need for a PV.

**Comment by balpert89** [comment]
Hi, so what is a possible fix to run the ASB successfully on 3.9?
I tried creating a valid PVC after the cluster installation:

´´´
oc scale dc/asb-etcd --replicas=0 -n openshift-ansible-service-broker
oc volume dc asb-etcd --remove --confirm --name=etcd -n openshift-ansible-service-broker
oc delete pvc etcd -n openshift-ansible-service-broker
oc set volume dc/asb-etcd --add --name=etcd --claim-name=etcd --type pvc --claim-class=ssd --claim-size=1G -n openshift-ansible-service-broker
oc scale dc/asb-etcd --replicas=1 -n openshift-ansible-service-broker
´´´

This way I receive a proper PV and the asb-etcd pod can be deployed, but it results in the error:
```
oc logs -f asb-etcd-4-lhc69
2018-06-01 19:09:03.343352 I | etcdmain: etcd Version: 3.2.18
2018-06-01 19:09:03.348144 I | etcdmain: Git SHA: eddf599
2018-06-01 19:09:03.348152 I | etcdmain: Go Version: go1.9.2
2018-06-01 19:09:03.348167 I | etcdmain: Go OS/Arch: linux/amd64
2018-06-01 19:09:03.348175 I | etcdmain: setting maximum number of CPUs to 4, total number of available CPUs is 4
2018-06-01 19:09:03.359811 I | embed: listening for peers on http://localhost:2380
2018-06-01 19:09:03.359962 I | embed: listening for client requests on 0.0.0.0:2379
2018-06-01 19:09:03.360070 C | etcdmain: cannot access data directory: mkdir /data: permission denied
```

Best regards
balpert89
